_delphi_utils_python/delphi_utils/smooth.py

```diff
@@ -443,34 +448,35 @@ def savgol_impute(self, signal):
         ----------
         signal: np.ndarray
             A 1D signal to be imputed.
+        impute_order: int
+            The polynomial order of the fit used for imputation.
 
         Returns
         ----------
         signal_imputed: np.ndarray
             An imputed 1D signal.
         """
+        if impute_order > self.window_length:
+            raise ValueError("Impute order must be smaller than window length.")
+
         signal_imputed = np.copy(signal)
         for ix in np.where(np.isnan(signal))[0]:
             # Boundary cases
             if ix < self.window_length:
                 # At the boundary, a single value should just be extended
                 if ix == 1:
                     signal_imputed[ix] = signal_imputed[ix - 1]
-                # Reduce the polynomial degree if needed
-                elif ix == 2:
-                    signal_imputed[ix] = self.savgol_predict(
-                        signal_imputed[:ix], 1, -1
-                    )
-                # Otherwise, use savgol fitting on the largest window prior
+                # Otherwise, use savgol fitting on the largest window prior,
+                # reduce the polynomial degree if needed
```

Comment:
add a note why the reduction of the polynomial degree is needed.

Reply:
Done